gmp, mpfr, virtualenv build broken in windows (msys2)

A few days ago, in order to rebase a couple of my branches on an up-to-date master, I made a pull that forced me to rebuild my `deps` (maybe due to #12463?).

Since then, though, I have not been able to rebuild gmp. It fails during config with text like this near the end:
```
checking size of mp_limb_t... 0
configure: error: Oops, mp_limb_t doesn't seem to work
Makefile:1803: recipe for target 'build/gmp-6.0.0/config.status' failed
```
I also get this error on a fresh clone.

~~Some googling suggests something about having absolute paths in the configure line breaks things.~~ If I go into the source directory and do a bare `./configure`, it succeeds (and gives a size of `mp_limb_t` of 8).

Has anyone else seen this? I've looked a bit into how we might tweak the invocation of `configure` but the logic in `deps/Makefile` looks, um, complicated.

Here is the error I get:
```
"/c/tw/Anaconda/python" /c/tw/julia/deps/srccache/virtualenv-1.11.6/virtualenv.py build/julia-env
New python executable in build/julia-env\Scripts\python.exe
Installing setuptools, pip...done.
[ -e build/julia-env/Scripts ] && ! [ -e build/julia-env/bin ] && cmd //C mklink //J build/julia-env\\bin build/julia-env\\Scripts
Parameter format not correct - "julia-env\bin".
Makefile:1995: recipe for target 'build/julia-env' failed
make[3]: [build/julia-env] Error 1 (ignored)
touch -c build/julia-env
make[3]: Leaving directory '/c/tw/julia/deps'
touch -c ../deps/build/julia-env/bin/activate
. ./../deps/build/julia-env/bin/activate && pip install sphinx==1.3.1 \
              && pip install -r /c/tw/julia/doc/requirements.txt
/bin/sh: ./../deps/build/julia-env/bin/activate: No such file or directory
Makefile:33: recipe for target '../deps/build/julia-env/bin/sphinx-build' failed
make[2]: *** [../deps/build/julia-env/bin/sphinx-build] Error 1
make[2]: Leaving directory '/c/tw/julia/doc'
/c/tw/julia/Makefile:56: recipe for target '/c/tw/julia/doc/_build/html' failed
make[1]: *** [/c/tw/julia/doc/_build/html] Error 2
make[1]: Leaving directory '/c/tw/julia'
Makefile:438: recipe for target 'binary-dist' failed
make: *** [binary-dist] Error 2
```

Unfortunately not (MSYS2).

```
checking size of mp_limb_t... 0
configure: error: Oops, mp_limb_t doesn't seem to work
Makefile:1795: recipe for target 'build/gmp-6.1.0/config.status' failed
```

Sorry, not yet. Made a ``make cleanall`` but still errors 

```
checking size of mp_limb_t... 0
configure: error: Oops, mp_limb_t doesn't seem to work
Makefile:1797: recipe for target 'build/gmp-6.1.0/config.status' failed
make[1]: *** [build/gmp-6.1.0/config.status] Error 1
Makefile:78: recipe for target 'julia-deps' failed
```

well make works, but make install still fails as above

    $ make install
    make[1]: Entering directory '/home/jeffw/src/julia/doc'
    make -C ./../deps install-virtualenv
    make[2]: Entering directory '/home/jeffw/src/julia/deps'
    "/c/Python27/python" /home/jeffw/src/julia/deps/srccache/virtualenv-1.11.6/virtualenv.py build/julia-env
    New python executable in build/julia-env\Scripts\python.exe
    Installing setuptools, pip...done.
    [ -e build/julia-env/Scripts ] && ! [ -e build/julia-env/bin ] && cmd //C mklink //J build/julia-env\\bin build/julia-env\\Scripts
    Parameter format not correct - "julia-env\bin".
    Makefile:1996: recipe for target 'build/julia-env' failed
    make[2]: [build/julia-env] Error 1 (ignored)
    touch -c build/julia-env
    make[2]: Leaving directory '/home/jeffw/src/julia/deps'
    touch -c ../deps/build/julia-env/bin/activate
    . ./../deps/build/julia-env/bin/activate && pip install sphinx==1.3.1 \
              && pip install -r /home/jeffw/src/julia/doc/requirements.txt
    /bin/sh: ./../deps/build/julia-env/bin/activate: No such file or directory
    Makefile:33: recipe for target '../deps/build/julia-env/bin/sphinx-build' failed
    make[1]: *** [../deps/build/julia-env/bin/sphinx-build] Error 1
    make[1]: Leaving directory '/home/jeffw/src/julia/doc'
    Makefile:56: recipe for target '/home/jeffw/src/julia/doc/_build/html' failed
    make: *** [/home/jeffw/src/julia/doc/_build/html] Error 2

--- a/deps/Makefile
+++ b/deps/Makefile
@@ -1995,7 +1995,7 @@ $(VIRTUALENV_SOURCE): $(SRCDIR)/srccache/virtualenv-$(VIRTUALENV_VER).tar.gz
 $(VIRTUALENV_TARGET): $(VIRTUALENV_SOURCE)
 	"$(shell $(SRCDIR)/find_python2)" $< $@
 ifeq ($(BUILD_OS), WINNT)
-	-[ -e $@/Scripts ] && ! [ -e $@/bin ] && cmd //C mklink //J $@\\bin $@\\Scripts
+	-[ -e $@/Scripts ] && ! [ -e $@/bin ] && cmd //C mklink //J `echo $@/bin $@/Scripts | sed -e 's#/#\\\\#g'`
 endif
 	touch -c $@
 

